tests/common/lib/harbor.bash

```diff
@@ -0,0 +1,141 @@
+
+harbor_load_env() {
+  harbor_endpoint="$(yq sc '.harbor.subdomain + "." + .global.baseDomain')"
+  export harbor_endpoint
+
+  harbor_username="admin"
+  export harbor_username
+
+  harbor_password="$(yq_secret ".harbor.password")"
+  export harbor_password
+}
+
+# Expects variables to be set with harbor_load_env
+harbor_setup_project() {
+  local output
+
+  export project="end-to-end-${1:-"user-demo"}-project"
+  export robot="end-to-end-${1:-"user-demo"}-robot"
+
+  export robot_id_path="/tmp/compliantkubernetes-apps-end-to-end-${1:-"user-demo"}-robot-id"
+  export robot_name_path="/tmp/compliantkubernetes-apps-end-to-end-${1:-"user-demo"}-robot-name"
+  export robot_secret_path="/tmp/compliantkubernetes-apps-end-to-end-${1:-"user-demo"}-robot-secret"
+
+  run harbor_create_project "${project}"
+
+  refute_output
+  assert_success
+
+  run harbor_create_robot "${project}" "${robot}"
+
+  assert_line --regexp ".*\"name\":\"robot.*"
+  assert_success
+
+  jq -r .id <<< "${output}" > "${robot_id_path}"
+  jq -r .name <<< "${output}" > "${robot_name_path}"
+  jq -r .secret <<< "${output}" > "${robot_secret_path}"
+
+  run docker login --username "$(<"${robot_name_path}")" --password-stdin "${harbor_endpoint}" < "${robot_secret_path}"
+
+  assert_line --regexp "Login Succeeded"
+  assert_success
+}
+
+# Expects a project to be setup with harbor_setup_project
+harbor_teardown_project() {
+  docker logout "${harbor_endpoint}"
+
+  read -r robot_id < "${robot_id_path}"
+  export robot_id
+
+  run harbor_delete_robot "${robot_id_path}"
+
+  refute_output
+  assert_success
+
+  rm -f "${robot_id_path}"
+  rm -f "${robot_name_path}"
+  rm -f "${robot_secret_path}"
+
+  run harbor_delete_project "${project}"
+
+  refute_output
+  assert_success
+}
+
+# Expects a project to be setup with harbor_setup_project
+harbor_setup_user_demo_image() {
+  export docs_repo="https://github.com/elastisys/compliantkubernetes.git"
+  export docs_path="/tmp/compliantkubernetes"
+
+  if [[ -d "${docs_path}/.git" ]]; then
+    pushd "${docs_path}" || exit 1
+    git checkout -- .
+    git switch main
+    git pull
+    popd || exit 1
+  else
+    git clone "${docs_repo}" "${docs_path}"
+  fi
+
+  export user_demo="${docs_path}/user-demo/"
+  export user_demo_image="${harbor_endpoint}/${project}/user-demo:test"
+
+  docker build "${user_demo}" -t "${user_demo_image}"
+  docker push "${user_demo_image}"
+}
+
+harbor_teardown_user_demo_image() {
+  run harbor_delete_repository "${project}" user-demo
+
+  refute_output
+  assert_success
+}
+
+harbor_get_current_user() {
+  curl -s -u "${harbor_username}:${harbor_password}" -X GET "https://${harbor_endpoint}/api/v2.0/users/current" \
+    -H "accept: application/json"
+}
+
+harbor_create_project() {
+  curl -s -u "${harbor_username}:${harbor_password}" -X POST "https://${harbor_endpoint}/api/v2.0/projects" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{\"project_name\": \"${1}\"}"
```

Comment:
```suggestion
    -d "$(jq --arg project_name "${1}" --null-input '{project_name: $project_name}')"
```
This kind of thing should make JSON literal arguments safer to build, tho in tests you could count on nobody ever deciding to put quotes or `\` in these variables...